From 194cf8a4446bb622e126245828253778d07fcf65 Mon Sep 17 00:00:00 2001
From: guoanwu <33250245+guoanwu@users.noreply.github.com>
Date: Wed, 11 Mar 2020 16:39:31 +0800
Subject: [PATCH 02/23] Update CONTRIBUTING

---
 CONTRIBUTING | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CONTRIBUTING b/CONTRIBUTING
index 2c74b939d..becd3e8a3 100644
--- a/CONTRIBUTING
+++ b/CONTRIBUTING
@@ -15,9 +15,9 @@ to accomplish and why. Use cases are important for features to be accepted.
 Here you'll see if there is consensus about your idea.
 
 2. If in step 1 you get an acknowledgment from the project leaders, use the
-   following procedure to submit a patch:
+   following procedure to submit a patch:D
 
-    a. Fork MemKeySB on github ( http://help.github.com/fork-a-repo/ )
+    a. Fork MemKeyDB on github ( http://help.github.com/fork-a-repo/ )
     b. Create a topic branch (git checkout -b general-change-topic)
     c. Push to your branch (git push origin general-change-topic)
     d. Initiate a pull request on github ( https://help.github.com/articles/creating-a-pull-request/ )
-- 
2.19.1.6.gb485710b

