From 209cde6bc2c86d9b6c58d8d058e8cd0665fe9538 Mon Sep 17 00:00:00 2001
From: michalbiesek <michal.biesek@intel.com>
Date: Tue, 14 Apr 2020 15:05:47 +0200
Subject: [PATCH 05/16] Use memkind allocator as default

---
 README.md    |  6 ++----
 src/Makefile | 11 ++---------
 2 files changed, 4 insertions(+), 13 deletions(-)

diff --git a/README.md b/README.md
index 9aaa229c2..3eb2827fe 100644
--- a/README.md
+++ b/README.md
@@ -58,10 +58,8 @@ Allocator
 ---------
 
 Selecting a non-default memory allocator when building MemKeyDB is done by setting
-the `MALLOC` environment variable. MemKeyDB is compiled and linked against libc
-malloc by default, with the exception of jemalloc being the default on Linux
-systems. This default was picked because jemalloc has proven to have fewer
-fragmentation problems than libc malloc.
+the `MALLOC` environment variable. MemKeyDB is compiled and linked against
+memkind malloc by default.
 
 To force compiling against libc malloc, use:
 
diff --git a/src/Makefile b/src/Makefile
index fe84dce48..9ce1573c8 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -34,15 +34,8 @@ INSTALL_BIN=$(PREFIX)/bin
 INSTALL=install
 PKG_CONFIG?=pkg-config
 
-# Default allocator defaults to Jemalloc if it's not an ARM
-MALLOC=libc
-ifneq ($(uname_M),armv6l)
-ifneq ($(uname_M),armv7l)
-ifeq ($(uname_S),Linux)
-	MALLOC=jemalloc
-endif
-endif
-endif
+# Default allocator defaults to memkind
+MALLOC=memkind
 
 # To get ARM stack traces if Redis crashes we need a special C flag.
 ifneq (,$(filter aarch64 armv,$(uname_M)))
-- 
2.19.1.6.gb485710b

